miden/tests/integration/stdlib/collections/smt.rs

```diff
@@ -0,0 +1,191 @@
+use super::build_test;
+use miden::{crypto::Rpo256, math::Felt};
+use vm_core::{
+    crypto::merkle::{MerkleStore, NodeIndex},
+    utils::IntoBytes,
+    StarkField, Word, WORD_SIZE,
+};
+
+#[test]
+fn mtree_get_works_for_subtree() {
+    let source = r#"
+        begin
+            mtree_get
+        end
+    "#;
```

Comment:
Seems like this test is testing only the `mtree_get` instruction - so, it probably should not be here?

Reply:
Indeed; all-in-all it is testing the ability of the `MerkleStore` to handle sub-trees. I added here because this is a requirement of the TSMT, but we can promote it to miden-crypto. Wdyt?

Reply:
Test case removed